### DISCUSSION TITLE
v2 beta README

### DISCUSSION THREAD

**Discussion by JordanMarr** [JordanMarr/ReactiveElmish.Avalonia#30] [Announcements]
I have been updating the v2 readme with the new features.

Looking for feedback if anything is less than clear. Especially if you have had any pain points while using the library around things that that may not be very clear. 

So far, I think the “Why” section is useful (as some people do ask that question, but the v1 vs V2 history section could possibly be removed. 

https://github.com/JordanMarr/Elmish.Avalonia/tree/v2-beta

**Comment by TeaDrivenDev** [comment]
No feedback on the content as such yet, but you appear to have deleted a few occurrences of "bind", so there's a bunch of headless "ings" running around now.... :D

**Comment by houstonhaynes** [comment]
I'm seeing some gaps in text that looks like markup might be mis-interpreted.

![Screenshot 2023-11-26 at 7 21 20 AM](https://github.com/JordanMarr/Elmish.Avalonia/assets/8174976/b555b571-cbe2-48f9-acd0-f04e5c200cc1)

**Reply by houstonhaynes** [reply]
To pick a nit - think "more powerful" might need a bit of unpacking... or re-wording to be a objective/specific.

> While the built-in F# templates do allow you to do classic MVVM style, Elmish is more powerful and has become the standard for F# UI projects.

[I honestly don't know whether that would open up an entirely new avenue of explanation and blow up the size of the text, so feel free to ignore this call-out]

**Reply by houstonhaynes** [reply]
I love this! I think it's going to have quite a bit of "ripple effect" out into the Reactive/Avalonia and F# thought spaces. 

Also, I can't help thinking that this deserves its own template. There's definitely value in transitioning from a "vanilla" MVVM template but I wonder if the number of touch points might be enough friction to dissuade people from using it. Does it make sense to have a "template" that is - in essence - the sample app? (and in the case where folks want to go XPlat - spawning that format with appropriate substitution of name lables?)

**Reply by JordanMarr** [reply]
I hope so! It feels pretty satisfying to use -- even fun (to me).

There is probably enough boiler plate to justify a template that recreates a minimal version of the sample application that includes routing, counter page and about page.

**Comment by JordanMarr** [comment]
Fixed the broken `ings` 😅 and reworded the "Elmish is more powerful" to something slightly less subjective: 
"While the built-in F# templates do allow you to do classic MVVM style, Elmish provides a powerful form of state management that has become standard for F# UI projects."

**Comment by kgday** [comment]
Looks good.  A couple of nit picks:

> Elmish.Avalonia provides custom integration for ReactiveUI.DynamicData which provides a very simple way to lists between the Elmish model and the view / view model.
".. a simply way to manage lists between the .." maybe? Or ".. a simply way to work with lists between the .."

Also:

> The BindSourceList method binds a ReactiveUI [SourceList](https://www.reactiveui.net/docs/handbook/collections)

SourceList is define in DynamicData not ReactiveUI.

**Comment by TeaDrivenDev** [comment]
At this point I'd probably do away with the whole "History" part; it takes a lot of space and distracts from the important things. If you want to keep it as a reference, it could be condensed to something like "While the original Elmish.Avalonia library was a direct port of the Elmish.WPF project, ReactiveElmish.Avalonia takes an entirely new approach."

**Reply by JordanMarr** [reply]
Agreed.
